docker/dependencies/Dockerfile

```diff
@@ -0,0 +1,298 @@
+    cp /tmp/tbb/build/build_release/*.so* /opt/dependencies/tbb/lib; \
+    cp -r /tmp/tbb/include/* /opt/dependencies/tbb/include; \
+    ldconfig; \
+    rm -rf /tmp/tbb
+
+# install sonar cli
+#ENV SONAR_CLI_VERSION=3.0.3.778
+#RUN set -e; \
+#    mkdir -p /opt/sonar; \
+#    curl -L -o /tmp/sonar.zip https://sonarsource.bintray.com/Distribution/sonar-scanner-cli/sonar-scanner-cli-${SONAR_CLI_VERSION}-linux.zip; \
+#    unzip -o -d /tmp/sonar-scanner /tmp/sonar.zip; \
+#    mv /tmp/sonar-scanner/sonar-scanner-${SONAR_CLI_VERSION}-linux /opt/sonar/scanner; \
+#    ln -s -f /opt/sonar/scanner/bin/sonar-scanner /usr/local/bin/sonar-scanner; \
+#    rm -rf /tmp/sonar*
+
+# install ed25519
+RUN set -e; \
+    git clone git://github.com/hyperledger/iroha-ed25519.git /tmp/ed25519; \
+    (cd /tmp/ed25519 ; git checkout e7188b8393dbe5ac54378610d53630bd4a180038); \
+    cmake \
+        -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE} \
+        -DTESTING=OFF \
+        -H/tmp/ed25519 \
+        -B/tmp/ed25519/build \
+        -DCMAKE_INSTALL_PREFIX=/opt/dependencies/ed25519; \
+    cmake --build /tmp/ed25519/build --target install -- -j${PARALLELISM}; \
+    ldconfig; \
+    rm -rf /tmp/ed25519
+
+# fetch lcov reports converter
+#RUN set -e; \
+#    curl -L -o /tmp/lcov_cobertura.py https://raw.githubusercontent.com/eriwen/lcov-to-cobertura-xml/8c55cd11f80a21e7e46f20f8c81fcde0bf11f5e5/lcov_cobertura/lcov_cobertura.py
+
+#RUN set -e; \
+#    add-apt-repository -y ppa:webupd8team/java; \
+#    apt-get update; \
+#    echo oracle-java8-installer shared/accepted-oracle-license-v1-1 select true | /usr/bin/debconf-set-selections; \
+#    apt-get -y install oracle-java8-installer; \
+#    java -version
+
+# Build SWIG
+#RUN set -e; \
+#    curl -L -o /tmp/swig-3.0.12.tar.gz https://github.com/swig/swig/archive/rel-3.0.12.tar.gz; \
+#    tar -C /tmp -zxf /tmp/swig-3.0.12.tar.gz; \
+#    cd /tmp/swig-rel-3.0.12; \
+#    ./autogen.sh && ./configure && make -j${PARALLELISM}; \
+#    make install; \
+#    rm -rf /tmp/swig-rel-3.0.12
+
+#RUN set -e; \
+#    add-apt-repository -y ppa:jonathonf/python-3.6; \
+#    apt-get update; \
+#    apt-get -y install python3.6-dev
+
+# python bindings dependencies
+#RUN set -e; \
+#    pip install grpcio_tools; \
+#    pip3 install grpcio_tools
+
+# install lcov
+#RUN set -e; \
+#    curl -L -o /tmp/lcov-1.13.tar.gz https://github.com/linux-test-project/lcov/releases/download/v1.13/lcov-1.13.tar.gz; cd /tmp; tar zxf lcov-1.13.tar.gz; cd lcov-1.13; make install
+
+# non-interactive adduser
+#   -m = create home dir
+#   -s = set default shell
+#   iroha-dpnds-test = username
+#   -u = userid, default for Ubuntu is 1000
+#   -U = create a group same as username
+#   no password
+RUN useradd -ms /bin/bash iroha-dpnds-test -u 1000 -U
+
+WORKDIR /opt/iroha
+RUN set -e; \
+    chmod -R 777 /opt/iroha; \
+    mkdir -p /tmp/ccache -m 777; \
+    ccache --clear
+
+# udpate necessary environment variables
+RUN set -e; \
```

Comment:
I have tried not to export the variables, as it seemed, DCMAKE_PREFIX_PATH is not enough for ld, which does not see some of the libraries, if they are not in some of the default directories or environment variables, such as these two

Reply:
Do you mean the following error?
```
/opt/dependencies/grpc/bin/grpc_cpp_plugin: error while loading shared libraries: libgrpc_plugin_support.so: cannot open shared object file: No such file or directory
--grpc_out: protoc-gen-grpc: Plugin failed with status code 127.
schema/CMakeFiles/yac_grpc.dir/build.make:68: recipe for target '../schema/yac.grpc.pb.h' failed
```
It is expected, and it should be handled in cmake.

Reply:
I cannot repeat an error here, but I got a lot of them during make, because some of the libraries could not be found by ld and thus linking ended with failure

Reply:
I suggest to try using separate variables for libraries, includes, etc. Such as `CMAKE_LIBRARY_PATH` or `CMAKE_INCLUDE_PATH`. Because `CMAKE_PREFIX_PATH` probably starts by looking for known directory names (e.g.,`lib`) which could be absent in the package's folder.

Reply:
@bakhtin I thought that installation of libraries implies the directory structure, so that it is defined that libraries are installed in `$PREFIX/lib`, includes are installed in `$PREFIX/include` and so on. Is there a dependency which does not follow this rule?
We will have to maintain separate lists for subdirectories manually if we use `CMAKE_LIBRARY_PATH` or `CMAKE_INCLUDE_PATH`, while with `CMAKE_PREFIX_PATH` we can generate the list with a `ls /opt/dependencies/* -d1 | paste -s -d\; -`.

Reply:
I can not seem to find in CMake docs which directories are scanned for libraries. But it definitely includes /lib, /usr/lib, /usr/${arch}/lib subdirectories.